pkg/virt-handler/vm.go

```diff
@@ -1119,6 +1119,15 @@ func (d *VirtualMachineController) updateVMIStatus(origVMI *v1.VirtualMachineIns
 		log.Log.Errorf("virt-launcher does not support the Secure Boot setting. Updating VMI %s status to Failed", vmi.Name)
 		vmi.Status.Phase = v1.Failed
 	}
+
+	if vmi.ResourceVersion == "" {
```

Comment:
Needs a test.

Reply:
Added the unit test for `It("should get latest resourceVersion before update if resourceVersion is missing", func() {...})`